spring-xd-rest-client/src/main/java/org/springframework/xd/rest/client/impl/StreamTemplate.java

```diff
@@ -95,4 +97,17 @@ public void destroyAll() {
 		restTemplate.delete(resources.get("streams/definitions").expand());
 	}
 
+	@Override
+	public void deleteQueue(String name, Map<String, String> properties) {
+		String rootUri = resources.get("streams/clean/rabbit").toString() + "/{name}";
+		UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(rootUri);
+
+		for(Map.Entry<String, String> entry: properties.entrySet()) {
+			builder.queryParam(entry.getKey(), entry.getValue());
+		}
```

Comment:
1. I am not sure what you mean; I meant add the ability to optionally configure the rabbitmq details once, during template creation (with a bunch of setters), and then merge the runtime properties with the configuration properties.

2. Fine with me (intermediate class); I simply abhor seeing essentially identical code in sibling classes - it's a maintenance nightmare.